### ISSUE TITLE
Define NodeExePath within MSBuild Target

### ISSUE THREAD

**Issue by crumblycake** [Microsoft/nodejstools#1598]
When the NodeExePath property is left undefined (and so relying on the default installed path) there doesn't appear to be any access for custom MSBuild rules to use the "default" value. 
Could this default value be set on the property (preferably somewhere **before** the Build target) so that it is possible for MSBuild targets to make use of it?
For our own case - in brief - we import a custom MSBuild .targets file to run the locally installed webpack, parse the output, fill out the VS Error List and "npm pack" the package if the build was successful.
Currently this is possible but it is made flimsy by implicit calls to node/npm (relying on PATH to be setup in the right order) or explicit calls (not good for source control).

**Comment by paulvanbrenk** [comment]
When the NodeExePath is left undefined we depend on the PATH to find the exe... we only display it in the UI to show the value we got. 

So if we would set the value, we would start to break users that depend on the value being the one in PATH, e.g. when they switch between machines with Node 32bit or Node 64 bit installed in either `Program Files` or `Program Files (x86)`.

**Comment by crumblycake** [comment]
Sorry I maybe should've been clearer. I meant setting the property (if it hasn't already been set) during just the execution phase (e.x. via the project itself before invoking msbuild or within an msbuild target). 
Something like:
```xml
<Target Name="SetupNodeExePath" BeforeTargets="Build">
    <GetPathToNodeExecutableFromEnvironment Condition="$(NodeExePath) == ''">
        <Output TaskParameter="NodeExePath" PropertyName="NodeExePath" />
    </GetPathToNodeExecutableFromEnvironment>
</Target>
```
Still upholds the idea of implicit values as it's only of that value during the appropriate execution phases and the GetPathToNodeExecutableFromEnvironment task could literally just be a wrapper to the equivalent method [Nodejs.GetPathToNodeExecutableFromEnvironment](https://github.com/Microsoft/nodejstools/blob/a537762e1718fa4e0bcce1c6fd8ab2030c9c8364/Nodejs/Product/Nodejs/Nodejs.cs#L56) which searches thoroughly for the exe and the logic of which is what we currently (and redundantly) have in a custom msbuild task.

**Comment by paulvanbrenk** [comment]
I see.

However we currently don't ship any MSBuild tasks, so that would require a significant amount of work. Mostly around setup etc... 

Feel free to contribute the MSBuild task, and I'll figure out how to do the setup work.

**Comment by kant2002** [comment]
@paulvanbrenk am I correct, that in order to ship that task, somebody should do following:
1. Create new project for MSBuild task
2. Reference output of that task in the TargetsVsix project so task would be shipped there.
3. Make sure that build process correctly copy output of step 1 to TargetsVsix

**Comment by paulvanbrenk** [comment]
Yep, that would be the work required in this repo. Then in the VS repo, we would need to make sure it's actually referenced correctly and installed in the correct location. 

@kant2002 I'm no longer @MSFT so I can't help you with any of that.
